CRM/Member/Form/MembershipRenewal.php

```diff
@@ -339,6 +339,12 @@ public function buildQuickForm() {
 
     $this->applyFilter('__ALL__', 'trim');
 
+    // CRM-20571
+    // Get the Join Date from Membership info as it is not available in the Renew form
+    $joinDate = CRM_Core_DAO::getFieldValue('CRM_Member_DAO_Membership', $this->_id, 'join_date');
+    // Save its value in a hidden field
+    $this->addElement('hidden', 'join_date', $joinDate);
```

Comment:
Yes, that is what I wanted to do. I asked on Stackexchange but nobody gave a hint about how to get a field from Membership form in Membership Renewal form. I saw similar code in Membership.php but didn't understand it fully and couldn't find the documentation. Didn't understand from where `$self` appeared. Can you point me to the documentation. I can then make that change. **Is there some place where I can find explanation of each parameter of each function in CiviCRM, probably with some examples.** The best documentation I could find was [QuickForm Reference](https://wiki.civicrm.org/confluence/display/CRMDOC/QuickForm+Reference) which just mentioned how `formRule()` works and the only argument which can be accessed in it is `$values`.